gnd/src/main/java/com/google/android/gnd/MainViewModel.java

```diff
@@ -149,12 +169,43 @@ void onSignedOut() {
     navigator.navigate(SignInFragmentDirections.showSignInScreen());
   }
 
-  private void onSignedIn() {
+  private Observable<NavDirections> onSignedIn() {
```

Comment:
Could you please create a `MainViewModelTest` and add unit tests for the new behaviors?

Reply:
updated https://github.com/google/ground-android/pull/845/commits/0dfe046871b1f70cf1d2a736aa110f157a4a34d9